### ISSUE TITLE
Feature request: <title> text separator characters

### ISSUE THREAD

**Issue by holloway** [dequelabs/axe-core#1823]
Often websites have their page `<title>` expressed with separator characters for the section of their site. Eg. there might HTML of:

    <title>Company | Contact Us | Email</title>

Note the `|` characters. The `<title>` element can only include Text Nodes, and it can't include HTML, so there's no possibility of hiding presentational characters from screen readers by having HTML such as this:

    <title>
      Company     <span aria-hidden="true">|</span>
      Contact Us  <span aria-hidden="true">|</span>
      Email
    </title>

The `|` characters as separators are spoken by most screen readers as "VERTICAL LINE" so the choice of separator characters is somewhat important, as I think being spoken "Company VERTICAL LINE Contact Us VERTICAL LINE Email" is probably not a good experience, and at least should warrant a warning.

This GitHub issue is asking whether an _**aXe warning**_ should be raised when we match characters that aren't suitable as separators, and the recommendation could be to consider alternatives such as the en-dash `-` which treated as a silent pause by all screen readers that I've tested.

Arguably the use of `|` is an example of [ASCII Art which is mentioned in WCAG 2.1](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content) and of course `|` isn't the only problematic choice because across the web there are many _artful_ separator characters such as `>`, `<`, `«`, `»`, and `•`. Indeed, the Deque.com homepage itself uses `|` as a separator character.

In terms of implementation and programming the rule could match a list of characters that are likely to be ASCII Art rather than content. Or perhaps we could invert that logic and match a regex of `[^\w\s-]`. There are many ways of programming this.

## Further reading
* [(2004) Title text separators](http://www.standards-schmandards.com/2004/title-text-separators/)
* [(2014) Deque puncuation and typographic symbols](https://www.deque.com/blog/dont-screen-readers-read-whats-screen-part-1-punctuation-typographic-symbols/)

**Comment by WilcoFiers** [comment]
I'm not too sure this is suitable for axe-core. I don't think it's a best practice to avoid the use of such characters in the page title. As you point out, even in the accessibility field this isn't uncommon. While I certainly see your point that this can be somewhat confusing, it is also such common place that doing something else can equally be confusing.

The other thing is that screen readers will often ignore punctuations like this, so users who don't want to hear this have a way to turn this off.

My suggestion would be that if you are looking to test for this to create a custom rule for this. You can do this using the [`axe.configure()` API](https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#api-name-axeconfigure).

**Comment by holloway** [comment]
@WilcoFiers I agree that due to the pervasiveness of `|` (and other ASCII art separators) it's not confusing for screen readers to hear "VERTICAL LINE", but I think it's also not a good experience and I would like to know what characters we should use. Do you know who in the industry might advise on such a thing? I have found very little discussion of this online.

Of course whatever the outcome of that choice of characters I'm keen to automate this test (either as default or a custom rule).

**Comment by WilcoFiers** [comment]
I played around with the idea a little more. For anyone interested the following custom rule should do what you're looking for:

```js
axe.configure({
  rules: [{
    id: 'no-title-separator',
    selector: 'title',
    excludeHidden: false,
    tags: ['best-practice'],
    metadata: {
      description:
        'Ensure titles are readable by screenreaders by avoiding separator characters',
      help: 'Do not use separator characters in the page title'
    },
    all: ['no-title-separator'],
    any: [],
    none: []
  }],
  checks: [{
    id: 'no-title-separator',
    evaluate: node => !/[|<>«»•]/.test(node.textContent),
    metadata: {
      impact: 'serious',
      messages: {
        pass: 'the page has no separator characters in the title',
        fail: 'the page has separator characters in the title'
      }
    }
  }]
});
```

As for getting a conversation on this going, I'd recommend https://web-a11y.slack.com/.

I'm going to close this issue, as this isn't something since this isn't something we're going to pursue further. Thank you for raising this though.
